src/nodes/NodeConnectionManager.ts

```diff
@@ -1659,14 +1660,64 @@ class NodeConnectionManager {
       throw new nodesErrors.ErrorNodeConnectionManagerConnectionNotFound();
     }
     // Need to make an authenticate request here. Get the connection and RPC.
+    let reader: ReadableStreamDefaultReader | undefined;
+    let writer: WritableStreamDefaultWriter | undefined;
     try {
       const authenticateMessage =
         await this.authenticateNetworkForwardCallback(ctx);
       await withF([this.acquireConnectionInternal(nodeId)], async ([conn]) => {
-        await conn.rpcClient.methods.nodesAuthenticateConnection(
-          authenticateMessage,
+        const authStream =
+          await conn.rpcClient.methods.nodesAuthenticateConnection(ctx);
+        writer = authStream.writable.getWriter();
+        reader = authStream.readable.getReader();
+
+        // Write the forward authentication message from this node
+        await writer.write(authenticateMessage);
+        const forwardMessageResultPair = await utils.valueOrTimeout(
+          reader.read(),
+          ctx,
+        );
+        const forwardMessageResult:
+          | SuccessMessage
+          | NodesAuthenticateConnectionMessage = forwardMessageResultPair.value;
```

Comment:
It's really weird you have to apply the type here. Likely because you're not properly defining the types above with the `let` statement.

Also, with a reader you should *ALWAYS* check the `done` property. In this case you're assuming that `value` is defined here which isn't true. So always check that the done and if the stream finished early then throw an explicit protocol error.